packages/blade/src/components/Accordion/__tests__/Accordion.web.test.tsx

```diff
@@ -0,0 +1,196 @@
+    expect(getAllByRole('region', { hidden: true })[0]).toHaveAttribute('aria-hidden', 'false');
+
+    // test second external button to toggle second accordion item
+    await user.click(getByRole('button', { name: externalButton2 }));
+    await waitFor(() => expect(getByText(description2)).toBeVisible());
+
+    expect(getByRole('button', { name: button2 })).toHaveAttribute('aria-expanded', 'true');
+    expect(getAllByRole('region', { hidden: true })[1]).toHaveAttribute('aria-hidden', 'false');
+
+    // first item should be collapsed now
+    expect(getAllByRole('region', { hidden: true })[0]).toHaveAttribute('aria-hidden', 'true');
+
+    // test collapse external button to collapse all accordion items
+    await user.click(getByRole('button', { name: externalButtonCollapse }));
+    await waitFor(() => expect(trigger1).toHaveAttribute('aria-expanded', 'false'));
+
+    expect(getAllByRole('region', { hidden: true })[0]).toHaveAttribute('aria-hidden', 'true');
+
+    expect(getByRole('button', { name: button2 })).toHaveAttribute('aria-expanded', 'false');
+    expect(getAllByRole('region', { hidden: true })[1]).toHaveAttribute('aria-hidden', 'true');
+  });
+
+  it('should accept testID', () => {
+    const { getByTestId } = renderWithTheme(
+      <Accordion testID="accordion">
+        <AccordionItem
+          title="How can I setup Route?"
+          description="You can use Razorpay Route from the Dashboard or using APIs to transfer money to customers. You may also check our docs for detailed instructions."
+          testID="accordion-item-1"
+        />
+        <AccordionItem
+          title="How can I setup QR Codes?"
+          description="Just use Razorpay. You may also check our docs for detailed instructions. Please use the search functionality to ask your queries."
+          testID="accordion-item-2"
+        />
+        <AccordionItem title="How can I setup Subscriptions?" description="Just use Razorpay." />
+      </Accordion>,
+    );
+
+    getByTestId('accordion');
+    getByTestId('accordion-item-1');
+    getByTestId('accordion-item-2');
+  });
+
+  it('should pass general a11y', async () => {
+    const { getAllByRole } = renderWithTheme(
+      <Accordion>
+        <AccordionItem
+          title="How can I setup Route?"
+          description="You can use Razorpay Route from the Dashboard or using APIs to transfer money to customers. You may also check our docs for detailed instructions."
+        />
+        <AccordionItem
+          title="How can I setup QR Codes?"
+          description="Just use Razorpay. You may also check our docs for detailed instructions. Please use the search functionality to ask your queries."
+        />
+        <AccordionItem title="How can I setup Subscriptions?" description="Just use Razorpay." />
+      </Accordion>,
+    );
+
+    await assertAccessible(getAllByRole('button')[0]);
+    await assertAccessible(getAllByRole('region', { hidden: true })[0]);
```

Comment:
Probably be better to assert accessibility on the whole tree instead of single elements, otherwise missing ID bindings and references might not get picked up as errors. 

await assertAccessible(container);

Reply:
Done in https://github.com/razorpay/blade/commit/5362504ae49ec5bbbae577c5543de2e9ba15b3cc